src/axom/quest/ShapeMesh.hpp

```diff
@@ -0,0 +1,343 @@
+  static constexpr axom::IndexType NUM_TETS_PER_HEX = HexahedronType::NUM_TRIANGULATE;
+
+  /*!
+   * @brief Constructor with computational mesh in a conduit::Node.
+   *
+   * @param [in] runtimePolicy The run-time policy used for computations
+   * @param [in] allocatorId Allocator id for internal and scratch space.
+   *   It should be compatible with @c runtimePolicy.
+   *   If axom::INVALID_ALLOCATOR_ID is specified, it will be
+   *   replaced with the default allocator for @c runtimePolicy.
+   *   For good performance, especially on GPUs, fast memory pools
+   *   should be used.
+   * @param [in/out] bpMesh Blueprint mesh to shape into.
+   * @param [in] topoName Name of the Blueprint topology.  If empty,
+   *   use the first topology in @c bpMesh.
+   * @param [in] matsetName Name of the Blueprint material set.
+   *   If empty, use the first material set in @c bpMesh.
+   *
+   * Incoming mesh array data are assumed to be accessible by the
+   * runtime policy, or an error will result.  (However the data need
+   * not correspond to the allocator id.)
+   */
+  ShapeMesh(RuntimePolicy runtimePolicy,
+            int allocatorId,
+            conduit::Node& bpMesh,
+            const std::string& topoName = {},
+            const std::string& matsetName = {});
+
+#ifdef AXOM_USE_SIDRE
+  /*!
+   * @brief Constructor with computational mesh in a sidre::Group.
+
+   * @param [in] runtimePolicy The run-time policy used for computations
+   * @param [in] allocatorId Allocator id for internal and scratch space.
+   *   It should be compatible with @c runtimePolicy.
+   *   If axom::INVALID_ALLOCATOR_ID is specified, it will be
+   *   replaced with the default allocator for @c runtimePolicy.
+   *   For good performance, especially on GPUs, fast memory pools
+   *   should be used.
+   * @param [in/out] bpMesh Blueprint mesh to shape into.
+   * @param [in] topoName Name of the Blueprint topology.  If empty,
+   *   use the first topology in @c bpMesh.
+   * @param [in] matsetName Name of the Blueprint material set.
+   *   If empty, use the first material set in @c bpMesh.
+   *
+   * Incoming mesh array data are assumed to be accessible by the
+   * runtime policy, or an error will result.  (However the data need
+   * not correspond to the allocator id.)
+  */
+  ShapeMesh(RuntimePolicy runtimePolicy,
+            int allocatorId,
+            sidre::Group* bpMesh,
+            const std::string& topoName = {},
+            const std::string& matsetName = {});
+#endif
+
+  /*!
+   * @brief Runtime policy set in constructor.
+   */
+  RuntimePolicy getRuntimePolicy() const { return m_runtimePolicy; }
+
+  /*!
+   * @brief Allocator id set in constructor.
+   */
+  int getAllocatorID() const { return m_allocId; }
+
+  /*
+   * !@brief Return computational mesh as a sidre::Group if it has
+   * that form, or nullptr otherwise.
+  */
+  sidre::Group* getMeshAsSidre() { return m_bpGrpExt; }
```

Comment:
If we're guarding sidre, this needs to be guarded as well.
Please check that you can successfully build/run the code w/ `AXOM_ENABLE_SIDRE=OFF`

Reply:
It turns out I had to guard the whole class against non-sidre builds.  One important requirement is to allocate all data in the same memory pool, by using the same memory allocator through out.  The mechanism for this is `sidre::ConduitMemory`.